src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -396,7 +396,7 @@ newtype SignedTx = SignedTx { signedTx :: ByteString }
     deriving (Show, Eq, Generic)
 
 data TxWitness
-    = PublicKeyWitness ByteString
+    = PublicKeyWitness ByteString (Hash "signature")
```

Comment:
`Key 'AddressK XPub` would feel better instead of `ByteString`, will do in separate pr because of import cycles

Reply:
`XPub` would probably be sufficient here.